src/traces/carpet/axis_defaults.js

```diff
@@ -203,6 +203,11 @@ module.exports = function handleAxisDefaults(containerIn, containerOut, options)
     // but no, we *actually* want to coerce this.
     coerce('tickmode');
 
+    if(!containerOut.title.text) {
+        delete containerOut.title.font;
+        delete containerOut.title.offset;
+    }
```

Comment:
Is there some reason it doesn't work to invert this pattern? ie only coerce these attributes if we have a title, rather than always coercing them then deleting if there's no title?

There is a lot of this pattern above - coerce everything and then delete - and it's needed when a value provided to one of these other attributes affects the default of the controlling attribute - like if there's an explicit start line color or width we default to show the start line, so we need to coerce `startlinecolor` and `startlinewidth` before we can correctly coerce `startline` but then if start line was explicitly hidden we need to delete color and width. But I don't think that applies here.

Reply:
> Is there some reason it doesn't work to invert this pattern?

I don't think so, but I can't say for sure TBH. What could be a reason for checking for an empty title and deleting title attributes at the end? In theory, every call to `coerce` could clear `containerOut.title`. But since there's no comment on the "why", let's reverse the logic as you suggested and keep :crossed_fingers: we haven't broke anything. Does that sound like a plan?